src/components/ModalWithForm.js

```diff
@@ -18,6 +19,14 @@ export default class ModalWithForm extends Modal {
     return formValues;
   }
 
+  setLoading(isLoading, text) {
+    if (isLoading) {
+      this._modalButton.textContent = "Saving...";
+    } else {
+      this._modalButton.textContent = text;
+    }
+  }
+

```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`